core/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -936,9 +936,16 @@ object SQLPlanParser extends Logging {
     // BuildRight, LeftOuter, ((CEIL(cast(id1#1490 as double)) <= cast(id2#1496 as bigint))
     // AND (cast(id1#1490 as bigint) < CEIL(cast(id2#1496 as double))))
     // Get joinType and buildSide by splitting the input string.
-    val nestedLoopParameters = exprStr.split(",", 3)
-    val buildSide = nestedLoopParameters(0).trim
-    val joinType = nestedLoopParameters(1).trim
+    val nestedLoopParameters = exprStr.split(",", 3).map(_.trim)
+    val (buildSide, joinType) =
+      if(JoinType.allsupportedJoinType.contains(nestedLoopParameters(1))) {
+        // If exprString has the format: BuildRight, Inner
+        (nestedLoopParameters(0).trim, nestedLoopParameters(1).trim)
+      } else {
+        // If exprString has the format: Inner, BuildRight
+        // Note: This format is present in Photon Event logs
+        (nestedLoopParameters(1).trim, nestedLoopParameters(0).trim)
```

Comment:
this perhaps goes to my last comment, it seems we aren't really doing the full mapping with parameters and everything?  if we are going to convert it seems more like to convert it fully over to have some format if possible. If its not possible then we need something like shim layer to deal with.

Reply:
- By default any Photon specific parsing is not needed in most cases. Replacing the name by its equivalent Spark name is sufficient. 
- Incase, a Photon specific parser is needed, we define a custom parser (eg. added classes like `PhotonBroadcastNestedLoopJoinExecParser`)